### PR TITLE
Add possibility to override fileName and base directory of Keycloak Quarkus distribution ZIP archive

### LINKED ISSUE #24283: Add possibility to override fileName and base directory of Keycloak Quarkus distribution ZIP archive.
### Description

This can be used by downstream projects to tweak names and content of distribution easier way.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by pskopek**
Closes #24283

### CODE REVIEW

File: quarkus/dist/pom.xml [hunk 1]
```diff
@@ -60,7 +60,7 @@
     </dependencies>
 
     <build>
-        <finalName>keycloak-${project.version}</finalName>
+        <finalName>${dist.archive.file}</finalName>
```
Comment (ASzc) [inline quarkus/dist/pom.xml:63]: Worth noting that this only affects the name inside `target/`, the installed/deployed name still conforms to the maven repo specification. Although, it doesn't hurt either.

For example, with `-Ddist.archive.base.file.name=xyzzy`:

```
aszczucz@tyrfing ~/s/k/keycloak (zip_name_change)> ls -l quarkus/dist/target/*.zip
-rw-r--r--. 1 aszczucz aszczucz 166047756 Oct 25 08:22 quarkus/dist/target/xyzzy-999.0.0.zip
aszczucz@tyrfing ~/s/k/keycloak (zip_name_change)> ls -l ~/.m2/repository/org/keycloak/keycloak-quarkus-dist/999.0.0-SNAPSHOT/*.zip
-rw-r--r--. 1 aszczucz aszczucz 166047756 Oct 25 08:22 /home/aszczucz/.m2/repository/org/keycloak/keycloak-quarkus-dist/999.0.0-SNAPSHOT/keycloak-quarkus-dist-999.0.0-SNAPSHOT.zip
```

**Review by ASzc — COMMENTED** [review]
Seems like it works. I'm questioning if `build-helper-maven-plugin` is really needed here, since maven is capable of concatenation of property values by itself:

```
    <properties>
        <dist.archive.base.file.name>keycloak</dist.archive.base.file.name>
        <dist.archive.base.dir.name>${dist.archive.base.file.name}</dist.archive.base.dir.name>
        <dist.archive.dir>${dist.archive.base.dir.name}-${project.version}</dist.archive.dir>
    </properties>
```

but it seems you want to trim any suffixes on the version number:

 ```
keycloak-999.0.0-SNAPSHOT/
vs
keycloak-999.0.0/
```

Perhaps a more generic thing to do here would be to use `build-helper-maven-plugin` to generate a `project.version.suffixless` or something, and then use that to let maven itself concat that with whatever we need, which is `dist.archive.base.dir.name` in this context. This would align with `project.version.npm` which was added recently for a different purpose.

**Comment by ASzc** [comment]
Although now that I think more about `project.version.npm`, the approach we arrived at there was to generate the value statically at the time project.version is updated (in `set-version.sh`), and not to try to do it in maven

File: quarkus/dist/pom.xml (hunk 1 above)
Comment (pskopek) [reply quarkus/dist/pom.xml:63]: I think this is OK to have different name in target/ . Processes using maven artifact download has to deal with it. The archive internal base dir will be correct.

**Review by stianst — APPROVED** [review]

**Pull request merged**